core/domain/suggestion_services.py

```diff
@@ -412,6 +413,39 @@ def accept_suggestion(
         _update_user_proficiency(user_proficiency)
 
 
+def update_suggestion(suggestion_id, translation_html):
+    """Updates a suggestion with the given suggestion_id and translation_html.
+
+    Args:
+        suggestion_id: str. The id of the suggestion to be accepted.
+        translation_html: str. The new translation html.
+
+    Raises:
+        Exception. The suggestion is already handled.
+        Exception. The suggestion is not valid.
+
+    Returns:
+        _update_suggestion(suggestion). Updates the suggestion.
+    """
+
+    suggestion = get_suggestion_by_id(suggestion_id)
+
+    if suggestion is None:
+        raise utils.ValidationError(
+            'You cannot change the suggestion with id %s because it does '
+            'not exist.' % (suggestion_id)
+        )
+    if suggestion.is_handled:
+        raise utils.ValidationError(
+            'The suggestion with id %s has already been accepted/'
+            'rejected.' % (suggestion_id)
```

Comment:
I think we should allow contributors to update the rejected suggestion i.e, a reviewer rejected a suggestion and the author should be responsible to update the suggestion with the review comments, right?

Reply:
I will add this feature if it is necessary after having more insights from Sagang.

Reply:
Yes, agreed that contributors should be able to update their suggestion, but this PR is focused purely on the reviewer's editing capability.